### PR TITLE
Update ModelServingGlobal Cypress test to account for sorting and filtering

### PR DESCRIPTION
**Pull request by ashley-o0o**
Closes: [RHOAIENG-3837](https://issues.redhat.com/browse/RHOAIENG-3837)

## Description
This PR validates that the Name and Project table filtering, searching, and sorting functionality works properly

## How Has This Been Tested?
`npm run test`

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/__tests__/cypress/cypress/pages/modelServing.ts [hunk 1]
```diff
@@ -58,6 +58,30 @@ class ModelServingGlobal {
...
+
+  findEmptyResults() {
+    return cy.findByTestId('no-result-found-title');
+  }
+
+  findSortButton(name: string) {
+    return this.modelServing().find('thead').findByRole('button', { name });
+  }
+
+  private modelServing() {
```
Comment (pnaik1) [inline frontend/src/__tests__/cypress/cypress/pages/modelServing.ts:82]: No need of this, can make use of `findModelsTable()`

**Review by pnaik1 — CHANGES_REQUESTED** [review]

File: frontend/src/__tests__/cypress/cypress/e2e/modelServing/ModelServingGlobal.cy.ts [hunk 2]
```diff
@@ -572,4 +574,58 @@ describe('Model Serving Global', () => {
...
+      initIntercepts({});
+      modelServingGlobal.visit('test-project');
+
+      // Verify initial run rows exist
+      modelServingGlobal.getModelRow('Test Inference Service').should('have.length', 1);
+
+      // Select the "Name" filter
+      TableToolbar.prototype.findFilterMenuOption('filter-dropdown-select', 'Name').click();
+      TableToolbar.prototype.findSearchInput().type('Test Inference Service');
+      // Verify only rows with the typed run name exist
```
Comment (ppadti) [inline frontend/src/__tests__/cypress/cypress/e2e/modelServing/ModelServingGlobal.cy.ts:589]: Can you target a specific table on this page?

File: frontend/src/__tests__/cypress/cypress/e2e/modelServing/ModelServingGlobal.cy.ts [hunk 3]
```diff
@@ -572,4 +574,58 @@ describe('Model Serving Global', () => {
...
+        projectEnableModelMesh: true,
+      });
+      modelServingGlobal.visit('test-project');
+
+      // Verify initial run rows exist
+      modelServingGlobal.getModelRow('Test Inference Service').should('have.length', 1);
+
+      // Select the "Project" filter
+      TableToolbar.prototype.findFilterMenuOption('filter-dropdown-select', 'Project').click();
+      TableToolbar.prototype.findSearchInput().type('test project');
```
Comment (ppadti) [inline frontend/src/__tests__/cypress/cypress/e2e/modelServing/ModelServingGlobal.cy.ts:615]: same here.

Comment (christianvogt) [reply frontend/src/__tests__/cypress/cypress/e2e/modelServing/ModelServingGlobal.cy.ts:615]: Looks like `TableToolbar` isn't correctly scoping its selectors. Please update at the same time.

**Pull request merged**